File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/PutReleasePipelineExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PutReleasePipelineExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var phases1Audiences1ConfigurationReleaseGuardianConfiguration = new ReleaseGuardianConfiguration();
        phases1Audiences1ConfigurationReleaseGuardianConfiguration.monitoringWindowMilliseconds(60000L);
        phases1Audiences1ConfigurationReleaseGuardianConfiguration.rolloutWeight(50);
        phases1Audiences1ConfigurationReleaseGuardianConfiguration.rollbackOnRegression(true);
        phases1Audiences1ConfigurationReleaseGuardianConfiguration.randomizationUnit("user");

        var phases1Audiences1Configuration = new AudienceConfiguration();
        phases1Audiences1Configuration.releaseStrategy("releaseStrategy_string");
        phases1Audiences1Configuration.requireApproval(true);
        phases1Audiences1Configuration.notifyMemberIds(List.of (
            "1234a56b7c89d012345e678f"
        ));
        phases1Audiences1Configuration.notifyTeamKeys(List.of (
            "example-reviewer-team"
        ));
        phases1Audiences1Configuration.releaseGuardianConfiguration(phases1Audiences1ConfigurationReleaseGuardianConfiguration);

        var phases1Audiences1 = new AudiencePost();
        phases1Audiences1.environmentKey("environmentKey_string");
        phases1Audiences1.name("name_string");
        phases1Audiences1.segmentKeys(List.of ());
        phases1Audiences1._configuration(phases1Audiences1Configuration);

        var phases1Audiences = new ArrayList<AudiencePost>(List.of (
            phases1Audiences1
        ));

        var phases1 = new CreatePhaseInput();
        phases1.name("Phase 1 - Testing");
        phases1.audiences(phases1Audiences);

        var phases = new ArrayList<CreatePhaseInput>(List.of (
            phases1
        ));

        var updateReleasePipelineInput = new UpdateReleasePipelineInput();
        updateReleasePipelineInput.name("Standard Pipeline");
        updateReleasePipelineInput.description("Standard pipeline to roll out to production");
        updateReleasePipelineInput.tags(List.of (
            "example-tag"
        ));
        updateReleasePipelineInput.phases(phases);

        try
        {
            var response = new ReleasePipelinesBetaApi(config).putReleasePipeline(
                "projectKey_string", // projectKey
                "pipelineKey_string", // pipelineKey
                updateReleasePipelineInput
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling ReleasePipelinesBetaApi#putReleasePipeline");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/PutReleasePipelineExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class PutReleasePipelineExample
{
    fun putReleasePipeline()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val phases1Audiences1ConfigurationReleaseGuardianConfiguration = ReleaseGuardianConfiguration(
            monitoringWindowMilliseconds = 60000,
            rolloutWeight = 50,
            rollbackOnRegression = true,
            randomizationUnit = "user",
        )

        val phases1Audiences1Configuration = AudienceConfiguration(
            releaseStrategy = null,
            requireApproval = true,
            notifyMemberIds = listOf (
                "1234a56b7c89d012345e678f",
            ),
            notifyTeamKeys = listOf (
                "example-reviewer-team",
            ),
            releaseGuardianConfiguration = phases1Audiences1ConfigurationReleaseGuardianConfiguration,
        )

        val phases1Audiences1 = AudiencePost(
            environmentKey = null,
            name = null,
            segmentKeys = listOf (),
            configuration = phases1Audiences1Configuration,
        )

        val phases1Audiences = arrayListOf<AudiencePost>(
            phases1Audiences1,
        )

        val phases1 = CreatePhaseInput(
            name = "Phase 1 - Testing",
            audiences = phases1Audiences,
        )

        val phases = arrayListOf<CreatePhaseInput>(
            phases1,
        )

        val updateReleasePipelineInput = UpdateReleasePipelineInput(
            name = "Standard Pipeline",
            description = "Standard pipeline to roll out to production",
            tags = listOf (
                "example-tag",
            ),
            phases = phases,
        )

        try
        {
            val response = ReleasePipelinesBetaApi().putReleasePipeline(
                projectKey = null,
                pipelineKey = null,
                updateReleasePipelineInput = updateReleasePipelineInput,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling ReleasePipelinesBetaApi#putReleasePipeline")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling ReleasePipelinesBetaApi#putReleasePipeline")
            e.printStackTrace()
        }
    }
}
